src/Symfony/Component/Cache/Adapter/RedisAdapter.php

```diff
@@ -229,53 +229,70 @@ protected function doSave(array $values, $lifetime)
                 $failed[] = $id;
             }
         }
-
         if (!$serialized) {
             return $failed;
         }
-        if ($lifetime > 0) {
-            if ($this->redis instanceof \RedisArray) {
-                $redis = array();
-                foreach ($serialized as $id => $value) {
-                    if (!isset($redis[$h = $this->redis->_target($id)])) {
-                        $redis[$h] = $this->redis->_instance($h);
-                        $redis[$h]->multi(\Redis::PIPELINE);
-                    }
-                    $redis[$h]->setEx($id, $lifetime, $value);
-                }
-                foreach ($redis as $h) {
-                    if (!$h->exec()) {
-                        $failed = false;
-                    }
-                }
-            } else {
-                $this->pipeline(function ($pipe) use ($serialized, $lifetime) {
-                    foreach ($serialized as $id => $value) {
-                        $pipe->setEx($id, $lifetime, $value);
-                    }
-                });
-            }
-        } elseif (!$this->redis->mSet($serialized)) {
-            return false;
+        if (0 >= $lifetime) {
+            $this->redis->mSet($serialized);
+
+            return $failed;
```

Comment:
this will not return an array rather than ``false``, and does not thech the return value from ``mSet``. Is it expected ?